server/services/v1/auth/gotrue.go

```diff
@@ -0,0 +1,601 @@
+		return nil, errors.Internal("Failed to list applications: reason = %s", err.Error())
+	}
+
+	// get admin access token
+	adminAccessToken, _, err := getGotrueAdminAccessToken(ctx, g)
+	if err != nil {
+		return nil, err
+	}
+
+	// make external call
+	getUsersUrl := fmt.Sprintf("%s/admin/users?created_by=%s&tigris_namespace=%s&page=1&per_page=5000", g.AuthConfig.GotrueURL, currentSub, currentNamespace)
+	client := &http.Client{}
+	getUsersReq, err := http.NewRequestWithContext(ctx, http.MethodGet, getUsersUrl, nil)
+	if err != nil {
+		log.Err(err).Msg("Failed to form request to delete user from gotrue")
+		return nil, errors.Internal("Failed to form request to get users request")
+	}
+	getUsersReq.Header.Add("Authorization", fmt.Sprintf("bearer %s", adminAccessToken))
+
+	getUsersResp, err := client.Do(getUsersReq)
+	if err != nil {
+		log.Err(err).Msg("Failed to get users from gotrue")
+		return nil, errors.Internal("Failed to get users from gotrue")
+	}
+
+	if getUsersResp.StatusCode != http.StatusOK {
+		log.Error().Int("status", getUsersResp.StatusCode).Msg("Received non OK status code to get users")
+		return nil, errors.Internal("Received non OK status code to get users")
+	}
+
+	// remove it from metadata cache
+	defer getUsersResp.Body.Close()
+
+	getUsersRespBytes, err := io.ReadAll(getUsersResp.Body)
+	if err != nil {
+		log.Err(err).Msg("Failed to read get users response")
+		return nil, errors.Internal("Failed to read get users response")
+	}
+	var getUsersRespJSON map[string]jsoniter.RawMessage
+	err = jsoniter.Unmarshal(getUsersRespBytes, &getUsersRespJSON)
+	if err != nil {
+		log.Err(err).Msg("Failed to parse getUsersResp")
+		return nil, errors.Internal("Failed to parse getUsersResp")
+	}
+
+	var users []map[string]jsoniter.RawMessage
+	err = jsoniter.Unmarshal(getUsersRespJSON["users"], &users)
+	if err != nil {
+		log.Err(err).Msg("Failed to parse getUsersResp - users")
+		return nil, errors.Internal("Failed to parse getUsersResp - users")
+	}
+
+	appKeys := make([]*api.AppKey, len(users))
+	for i, user := range users {
+		var email, name, description, createdBy, tigrisProject, clientSecret string
+		err := jsoniter.Unmarshal(user["email"], &email)
+		if err != nil {
+			log.Err(err).Msg("Failed to parse getUsersResp - email")
+			return nil, errors.Internal("Failed to parse getUsersResp - email")
+		}
+		clientId := strings.Split(email, "@")[0]
+
+		err = jsoniter.Unmarshal(user["encrypted_password"], &clientSecret)
+		if err != nil {
+			log.Err(err).Msg("Failed to parse getUsersResp - clientSecret")
+			return nil, errors.Internal("Failed to parse getUsersResp - clientSecret")
+		}
+
+		var appMetadata map[string]jsoniter.RawMessage
+		err = jsoniter.Unmarshal(user["app_metadata"], &appMetadata)
+		if err != nil {
+			log.Err(err).Msg("Failed to parse getUsersResp - appMetadata")
+			return nil, errors.Internal("Failed to parse getUsersResp - appMetadata")
+		}
+
+		err = jsoniter.Unmarshal(appMetadata["name"], &name)
```

Comment:
Why not define a small struct to decode from appMetadata, instead of deserializing individual fields?

Reply:
added small struct.